forte/base_classes/orbital_transform.cc

```diff
@@ -49,19 +54,32 @@ make_orbital_transformation(const std::string& type, std::shared_ptr<SCFInfo> sc
 
     if (type == "LOCAL") {
         orb_t = std::make_unique<Localize>(options, ints, mo_space_info);
-    }
-    if (type == "MP2NO") {
+    } else if (type == "MP2NO") {
         orb_t = std::make_unique<MP2_NOS>(scf_info, options, ints, mo_space_info);
-    }
-    if (type == "CINO") {
+    } else if (type == "CINO") {
         orb_t = std::make_unique<CINO>(scf_info, options, ints, mo_space_info);
-    }
-    if (type == "MRCINO") {
+    } else if (type == "MRCINO") {
         orb_t = std::make_unique<MRCINO>(scf_info, options, ints, mo_space_info);
-    }
+    } else if (type == "MRPT2NO") {
```

Comment:
I would say that this works as a temporary solution. Ideally all of this would be done on the python side and we would have some integral transform object that takes the current MOs and some other object that specifies how to transform the orbitals. I think that except for localize, all other methods have something in common: they diagonalize a matrix. This suggest that we could potentially split these two methods in the futures into two subclasses.

Reply:
I agree.